### DISCUSSION TITLE
[Feature Request] Add show query parameter to url

### DISCUSSION THREAD

**Discussion by CrazyBoy49z** [Power-Components/livewire-powergrid#360] [Ideas]
### Summary

How to correctly add filter queries to url, so you can save filtering

### Why is this needed?

You must save filtering requests when refreshing the page and navigating through the url by filtering

### Suggested Solution(s)

url?filters[is_banned]=true

Search all banned user

**Comment by luanfreitasdev** [comment]
This is very good but in the case of powergrid the best way to make this happen is to use dot notation along with alias. That's why I opened a PR on livewire. Let's wait!

https://github.com/livewire/livewire/pull/4913

**Reply by luanfreitasdev** [reply]
Currently the filter is not working properly but in v3 it will work normally.

That would be an example:
```php
public function getQueryString()
{
      return [
         'filters' => ['as' => $this->tableName . '_filters'],
      ];
}
```
Result:
`/filters?default_filters[multi_select][category_id][0]=2&default_filters[multi_select][category_id][1]=4`

The best scenario would be to generate an alias for each filter type:
```php
'filters.input_text.dishes.name'              => ['as' => $tableName . 'name', 'except' => ''],
'filters.input_text_options.dishes.name'      => ['as' => $tableName . 'name_options', 'except' => ''],
'filters.input_text_options.dishes.chef_name' => ['as' => $tableName . 'chef_name_options', 'except' => ''],
'filters.number.price.start'                  => ['as' => $tableName . 'number_start', 'except' => ''],
'filters.number.price.end'                    => ['as' => $tableName . 'number_end', 'except' => ''],
'filters.select.dishes.diet'                  => ['as' => $tableName . 'select', 'except' => ''],
'filters.multi_select.category_id'            => ['as' => $tableName . 'category_ids', 'except' => ''],
```

Result:
`filters?default_category_ids[0]=2&default_category_ids[1]=3&default_select=1
`

**Comment by kolydart** [comment]
What is the current state of the Feature Request?
Livewire 3 is already here for a long time. 
Can we implement this?
Filtering save is very important in many cases.
thanks
